project/Build.scala

```diff
@@ -1005,6 +1004,34 @@ object Build {
       ) ++ (
           scalaJSExternalCompileSettings
       ) ++ inConfig(Compile)(Seq(
+          scalacOptions in doc ++= Seq("-implicits", "-groups"),
+
+          // Filter doc sources to remove implementation details from doc.
+          sources in doc := {
+            def regexFileFilter(s: String): FileFilter = new FileFilter {
+              private val pat = s.r.pattern
+              override def accept(f: File): Boolean = {
+                val path = f.getAbsolutePath.replace('\\', '/')
+                pat.matcher(path).matches()
+              }
+            }
+
+            val filter: FileFilter = (
+                AllPassFilter
+                  -- regexFileFilter(".*/scala/scalajs/runtime/.*")
+                  -- regexFileFilter(".*/scala/scalajs/js/annotation/internal/.*")
+                  -- "*.nodoc.scala"
+            )
+
+            (sources in doc).value.filter(filter.accept)
+          },
+
+          /* Add compiled .class files to doc dependencyClasspath, so we can
+           * still compile even with only part of the files being present.
+           */
+          dependencyClasspath in doc +=
+            Def.task(classDirectory.value).dependsOn(compile).value,
```

Comment:
Note that this pulls in all the other library projects as well (scalalib, etc.). This is not an issue, but makes building the doc a bit slower than necessary.